oauthproxy_test.go

```diff
@@ -270,9 +270,9 @@ func (tp *TestProvider) ValidateSessionState(session *sessions.SessionState) boo
 	return tp.ValidToken
 }
 
-func (tp *TestProvider) ValidateGroup(email string) bool {
+func (tp *TestProvider) ValidateGroup(session *sessions.SessionState) bool {
```

Comment:
Why do we need to change the method signature here?

Reply:
Azure v2 api returns group membership information in id token. I'm using this in azure-oidc provider. Other approach would require parsing id token and storing that in some db under `email` key.

Reply:
What about renaming this function to ValidateUser. That way it would be simple to add more validations within the provider. I would for instance prefer to validate based on roles instead of groups. This feature could then be very easily added once this PR is merged.

The keycloak provider unfortunately does the group checking in the GetEmailAddress. I guess the reason there might have been, because that function gets the session as a parameter.